### PR TITLE
Move the symbol mapping service down to the Features layer.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide These are independent parts of the OOP PR that i'm separating out to simplify that review.

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_unit64_prtest  please

**Comment by CyrusNajmabadi** [comment]
corrupt nuget cache.

**Comment by CyrusNajmabadi** [comment]
Telling @MattGertz about this. Ths is part of the OOP work. It simplifies the large PR that is going through (#18939) by pulling out independent pieces and sending them through as much easier PRs.

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_unit64_prtest please

**Comment by CyrusNajmabadi** [comment]
retest windows_release_unit32_prtest please

**Comment by CyrusNajmabadi** [comment]
nuget cache corruption

**Comment by CyrusNajmabadi** [comment]
Retest windows_release_vs-integration_prtest  please

**Comment by CyrusNajmabadi** [comment]
navigateto flakeyness

**Comment by CyrusNajmabadi** [comment]
retest windows_release_vs-integration_prtest please

**Comment by CyrusNajmabadi** [comment]
Navigate to flakeyness.

**Comment by sharwell** [comment]
@CyrusNajmabadi I do appreciate the effort to break things into smaller reviews. Looking now. :smile:

**Comment by CyrusNajmabadi** [comment]
retest windows_release_vs-integration_prtest please

**Comment by CyrusNajmabadi** [comment]
nuget cache corruption

File: src/Features/Core/Portable/SymbolMapping/SymbolMappingServiceFactory.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.SymbolMapping
+{
+    [ExportWorkspaceService(typeof(ISymbolMappingService), ServiceLayer.Default), Shared]
+    internal class DefaultSymbolMappingService : ISymbolMappingService
```
Comment (sharwell) [inline src/Features/Core/Portable/SymbolMapping/SymbolMappingServiceFactory.cs]: :bulb: Would be good to have the type name match the file name here.

Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/SymbolMapping/SymbolMappingServiceFactory.cs]: Yup . I can make that change next.

**Pull request merged**